### PR TITLE
Add missing custom pass API documentation

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Add missing API doc in the custom passes

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs [hunk 1]
```diff
@@ -27,17 +27,23 @@ class Styles
...
 	    }
 
-		[Flags]
 		/// <summary>
 		/// List of the elements you can show/hide in the default custom pass UI.
 		/// </summary>
+		[Flags]
 		public enum PassUIFlag
 		{
+			/// <summary>Hide all the default UI fields</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs]: Hides all the default UI fields.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs [hunk 2]
```diff
@@ -27,17 +27,23 @@ class Styles
...
 			None				= 0x00,
+			/// <summary>Show the name field</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs]: Shows the name field.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs [hunk 3]
```diff
@@ -27,17 +27,23 @@ class Styles
...
 			Name				= 0x01,
+			/// <summary>show the target color buffer field</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs]: Shows the target color buffer field.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs [hunk 4]
```diff
@@ -27,17 +27,23 @@ class Styles
...
 			TargetColorBuffer	= 0x02,
+			/// <summary>Show the target depth buffer field</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs]: Shows the target depth buffer field.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs [hunk 5]
```diff
@@ -27,17 +27,23 @@ class Styles
...
 			TargetDepthBuffer	= 0x04,
+			/// <summary>Show the clear flags field</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs]: Shows the clear flags field.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs [hunk 6]
```diff
@@ -27,17 +27,23 @@ class Styles
...
 			ClearFlags			= 0x08,
+			/// <summary>Show all the default UI fields</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawer.cs]: Shows all the default UI fields.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawerAttribute.cs [hunk 7]
```diff
@@ -10,6 +10,10 @@ public class CustomPassDrawerAttribute : Attribute
 	{
 		internal Type targetPassType;
 
+		/// <summary>
+		/// Indicate that the class is a custom pass drawer and that it replace the default custom pass GUI.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawerAttribute.cs]: Indicates that the class is a custom pass drawer and that it replaces the default custom pass GUI.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawerAttribute.cs [hunk 8]
```diff
@@ -10,6 +10,10 @@ public class CustomPassDrawerAttribute : Attribute
...
+		/// </summary>
+		/// <param name="targetPassType">Your Custom Pass type</param>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassDrawerAttribute.cs]: The Custom Pass type.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 9]
```diff
@@ -90,8 +90,11 @@ internal ProfilingSampler   profilingSampler
         /// </summary>
         public enum TargetBuffer
         {
+            /// <summary>Current rendering camera buffers</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: The buffers for the currently rendering Camera.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 10]
```diff
@@ -90,8 +90,11 @@ internal ProfilingSampler   profilingSampler
...
             Camera,
+            /// <summary>Custom rendering buffers, allocated by HDRP</summary>
             Custom,
+            /// <summary>No buffer</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: No target buffer.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 11]
```diff
@@ -90,8 +90,11 @@ internal ProfilingSampler   profilingSampler
         /// </summary>
         public enum TargetBuffer
         {
+            /// <summary>Current rendering camera buffers</summary>
             Camera,
+            /// <summary>Custom rendering buffers, allocated by HDRP</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: The custom rendering buffers that HDRP allocates.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 12]
```diff
@@ -100,16 +103,27 @@ public enum TargetBuffer
         /// </summary>
         public enum RenderQueueType
         {
+            /// <summary>Opaque object without alpha test only</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: For this and every other entry in this section, could you please:
* Replace every 'object' with GameObject.
* Add a full stop at the end of every summary line.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 13]
```diff
@@ -100,16 +103,27 @@ public enum TargetBuffer
...
             AfterPostProcessOpaque,
+            /// <summary>Transparent objects with the pre refraction rendering pass</summary>
             PreRefraction,
+            /// <summary>Transparent objects with default rendering pass</summary>
             Transparent,
+            /// <summary>Transparent objects with low resolution rendering pass</summary>
             LowTransparent,
+            /// <summary>All Transparent objects</summary>
             AllTransparent,
+            /// <summary>Pre-refraction + Default + Low res transparent objects</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: Transparent GameObjects that use the Pre-refraction, Default, or Low resolution render pass.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 14]
```diff
@@ -174,8 +188,10 @@ internal void ExecuteInternal(ScriptableRenderContext renderContext, CommandBuff
 
         internal void InternalAggregateCullingParameters(ref ScriptableCullingParameters cullingParameters, HDCamera hdCamera) => AggregateCullingParameters(ref cullingParameters, hdCamera);
 
-        // Hack to cleanup the custom pass when it is unexpectedly destroyed, which happens every time you edit
-        // the UI because of a bug with the SerializeReference attribute.
+        /// <summary>
+        /// Hack to cleanup the custom pass when it is unexpectedly destroyed, which happens every time you edit
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: Cleans up the custom pass when Unity destroys it unexpectedly. Currently, this happens every time you edit

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 15]
```diff
@@ -356,7 +372,8 @@ protected RTHandle GetNormalBuffer()
         /// <summary>
         /// Returns the render queue range associated with the custom render queue type
         /// </summary>
-        /// <returns></returns>
+        /// <param name="type">The custom pass render queue type</param>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: The custom pass render queue type.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs [hunk 16]
```diff
@@ -356,7 +372,8 @@ protected RTHandle GetNormalBuffer()
...
+        /// <returns>a render queue range compatible with a ScriptableRenderContext.DrawRenderers</returns>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs]: Returns a render queue range compatible with a ScriptableRenderContext.DrawRenderers.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs [hunk 17]
```diff
@@ -13,11 +13,18 @@ public enum CustomPassInjectionPoint
     {
         // Important: don't touch the value of the injection points for the serialization.
         // Ordered by injection point in the frame for the enum popup in the UI.
+
+        /// <summary>Just after the depth clear, you can write to the depth buffer so Z-Tested opaque objects won't be rendered.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs]: Just after HDRP clears the depth buffer. You can write to the depth buffer to make HDRP not render depth-tested opaque GameObjects.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs [hunk 18]
```diff
@@ -13,11 +13,18 @@ public enum CustomPassInjectionPoint
...
         BeforeRendering             = 0,
+        /// <summary>At this point you can modify the normal, roughness and depth buffer, it will be taken in account in the lighting and the depth pyramid.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs]: At this point, you can modify the normal, roughness, and depth buffer. If you write to these buffers at this injection point, HDRP takes it into account in the lighting and the depth pyramid.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs [hunk 19]
```diff
@@ -13,11 +13,18 @@ public enum CustomPassInjectionPoint
...
         AfterOpaqueDepthAndNormal   = 5,
+        /// <summary>At this point to render any transparent objects that you want to be in the refraction (they, will end up in the color pyramid we use for refraction when drawing transparent objects)</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs]: At this injection point, you can render any transparent GameObject that you want to see in refraction. If you write to buffers at this injection point, they contents end up in the color pyramid that HDRP uses for refraction when it draws transparent GameObjects.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs [hunk 20]
```diff
@@ -13,11 +13,18 @@ public enum CustomPassInjectionPoint
...
         BeforePreRefraction         = 4,
+        /// <summary>At this point you can sample the color pyramid we generated for rough transparent refraction</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs]: At this injection point, you can sample the color pyramid that HDRP generates for rough transparent refraction.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs [hunk 21]
```diff
@@ -13,11 +13,18 @@ public enum CustomPassInjectionPoint
...
         BeforeTransparent           = 1,
+        /// <summary>Before the post process and custom post processes are rendered</summary>
         BeforePostProcess           = 2,
+        /// <summary>After the post processes. the depth is jittered so you can't draw depth tested objects without having artifacts</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs]: This injection point is after HDRP handles post-processesing. At this point, depth is jittered which means you cannot draw depth tested GameObjects without having artifacts.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs [hunk 22]
```diff
@@ -13,11 +13,18 @@ public enum CustomPassInjectionPoint
...
+
+        /// <summary>Just after the depth clear, you can write to the depth buffer so Z-Tested opaque objects won't be rendered.</summary>
         BeforeRendering             = 0,
+        /// <summary>At this point you can modify the normal, roughness and depth buffer, it will be taken in account in the lighting and the depth pyramid.</summary>
         AfterOpaqueDepthAndNormal   = 5,
+        /// <summary>At this point to render any transparent objects that you want to be in the refraction (they, will end up in the color pyramid we use for refraction when drawing transparent objects)</summary>
         BeforePreRefraction         = 4,
+        /// <summary>At this point you can sample the color pyramid we generated for rough transparent refraction</summary>
         BeforeTransparent           = 1,
+        /// <summary>Before the post process and custom post processes are rendered</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassInjectionPoint.cs]: This injection point is before HDRP renders post-processing and custom post-processing effects.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassVolume.cs [hunk 23]
```diff
@@ -240,6 +238,11 @@ internal static void Cleanup()
             }
         }
 
+        /// <summary>
+        /// Return the current active custom pass volume for a given injection point
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassVolume.cs]: Returns the currently active custom pass volume for a given injection point.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassVolume.cs [hunk 24]
```diff
@@ -240,6 +238,11 @@ internal static void Cleanup()
...
+        /// </summary>
+        /// <param name="injectionPoint">Custom pass injection point</param>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPassVolume.cs]: The injection point to get the currently active Custom Pass Volume for.

**Comment by JordanL8** [comment]
Added a lot of comments, if I've misunderstood anything or you have any questions, Slack me!

Also, Slack me when you've implemented the changes so I can approve the PR.

**Comment by alelievr** [comment]
I've fixed all the issues, I'll try to take notes for my future API doc changes :p

**Pull request merged**